test/avocado/testlib_avocado/machineslib.py

```diff
@@ -166,3 +167,87 @@ def tearDown(self):
         super().tearDown()
         if self.virshvm:
             self.destroy_vm(self.virshvm)
+        if len(self.vmlist) != 0:
+            for vm in self.vmlist:
+                self.destroy_vm(vm)
+        if self.machine.execute("sudo virsh pool-list | grep apool", raising=False) != '' and self.machine.execute("virsh pool-list | grep mypool", raising=False) != '':
+            self.machine.execute(
+                'sudo virsh vol-delete apooldisk1 apool \
+                    && sudo virsh vol-delete apooldisk2 apool \
+                    && sudo virsh vol-delete qcow2disk apool \
+                    && sudo virsh vol-delete raw2disk apool \
+                    && sudo virsh pool-destroy apool \
+                    && sudo rm -rf /home/apool \
+                    && sudo virsh vol-delete mypooldisk1 mypool \
+                    && sudo virsh vol-delete mypooldisk2 mypool \
+                    && sudo virsh pool-destroy mypool \
+                    && sudo rm -rf /home/mypool')
+        if self.machine.execute("sudo virsh vol-list default | grep detachdisk", raising=False) != '':
+            self.machine.execute('sudo virsh vol-delete detachdisk default')
+        if self.machine.execute('sudo cat /etc/passwd | grep auto', raising=False) != "":
+            self.machine.execute('sudo userdel -rf auto')
+
+    def prepare_disk(self):
+        self.machine.execute(
+            'sudo mkdir /home/{apool} \
+                && sudo virsh pool-create-as {apool} --type dir --target /home/{apool} \
+                && sudo virsh vol-create-as {apool} {apool}disk1 --capacity 1G --format qcow2 \
```

Comment:
maybe some underscore between the pool name and disk1/disk2 would be good for readiness.
Thus having name apool_disk1 instead of apooldisk1